libs/@local/hql/cst/src/symbol.rs

```diff
@@ -0,0 +1,292 @@
+use core::{fmt, fmt::Display};
+
+use smol_str::SmolStr;
+use unicode_ident::{is_xid_continue, is_xid_start};
+use winnow::{
+    combinator::{delimited, dispatch, empty, fail, opt, peek},
+    error::ParserError,
+    stream::{AsChar, Compare, Stream, StreamIsPartial},
+    token::{any, one_of, take_while},
+    PResult, Parser,
+};
+
+// TODO: in the future we might want to use the bump arena here as well.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct Symbol(SmolStr);
+
+impl Display for Symbol {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        Display::fmt(&self.0, f)
+    }
+}
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub(crate) enum ParseRestriction {
+    None,
+    RustOnly,
+}
```

Comment:
Could you add a small doc-string to describe what this means?

Reply:
added in [4004392](https://github.com/hashintel/hash/pull/4753/commits/400439251fa93808d90006ef2673b6d0e9feed08)